sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/action/permission/put/PutPermissionResponseInfo.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.action.permission.put;
+
+import org.opensearch.common.ParseField;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ConstructingObjectParser;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.transport.TransportResponse;
+
+import java.io.IOException;
+
+import static org.opensearch.common.xcontent.ConstructingObjectParser.constructorArg;
+
+/**
+ * Response info corresponds to a single response for a permission put action
+ */
+public class PutPermissionResponseInfo extends TransportResponse implements Writeable, ToXContent {
+    private final boolean successful;
+    private final String username;
+    private final String permissionString;
+
+    /**
+     * Construct an instance of a put permission response info (these are aggregated into a response for bulk requests)
+     * @param successful whether the request was successful
+     * @param username the username of the principal being granted the permission
+     * @param permissionString the permission string to be granted
+     */
+    public PutPermissionResponseInfo(boolean successful, String username, String permissionString) {
+        this.successful = successful;
+        this.username = username;
+        this.permissionString = permissionString;
+    }
+
+    /**
+     * A stream based constructor
+     * @param in an input stream for dealing with input from another node
+     * @throws IOException Throw on failure
+     */
+    public PutPermissionResponseInfo(StreamInput in) throws IOException {
```

Comment:
What is the use of this class? If a user calls the API with a list of permissions and at least 1 of the permissions in the request is invalid, but other permissions in the request are valid do you plan to fail the whole request or let the valid permissions be granted? IMO I think we should fail the whole request because I think it would be confusing for the user to have partial results when granting permissions.